src/stores/courseData.js

```diff
@@ -0,0 +1,232 @@
+import axios from 'axios';
+import moment from 'moment';
+import Vue from 'vue';
+import Vuex from 'vuex';
+
+Vue.use(Vuex);
+
+const DATE_FORMAT = 'YYYY-MM-DDTHH:mm:ss.SSS000Z';
+
+const store = new Vuex.Store({
+  strict: process.env.NODE_ENV !== 'production',
+  state: {
+    activeRoad: '$defaultroad$',
+    classInfoStack: [],
+    cookiesAllowed: false,
+    genericCourses: [],
+    genericIndex: {},
+    // note for later: will need to use Vue.set on roads for reactivity once they come from fireroad
+    roads: {
+      '$defaultroad$': {
+        downloaded: moment().format(DATE_FORMAT),
+        changed: moment().format(DATE_FORMAT),
+        name: 'My First Road',
+        agent: '',
+        contents: {
+          coursesOfStudy: ['girs'],
+          selectedSubjects: [],
+          progressOverrides: {}
+        }
+      }
+    },
+    subjectsIndex: {},
+    subjectsInfo: []
+  },
+  mutations: {
+    addClass (state, newClass) {
+      state.roads[state.activeRoad].contents.selectedSubjects.push(newClass);
+      Vue.set(state.roads[state.activeRoad], 'changed', moment().format(DATE_FORMAT));
```

Comment:
Do you mean you deleted the line entirely, or deleted the Vue.set part?  I think we can safely get rid of the Vue.set, but the changed property definitely needs to be set for it to work correctly with fireroad and sync to the server.

Reply:
I didn't mean that but in retrospect you're right. We can probably remove lots of `Vue.set`s then as long as their effects are preserved

Reply:
Yeah, I think a lot of them can be removed but we have to be careful because some are actually necessary (when manipulating arrays or adding new properties that weren't there before).

Reply:
Is this something we should do in this PR? I think fixing reactivity might be beyond the scope of this PR and be better left for another PR.

Reply:
Yeah let's leave that for another PR

Reply:
created #265 so we don't forget